bundles/admin/admin-layereditor/view/AdminLayerForm/PermissionsTabPane.jsx

```diff
@@ -1,7 +1,70 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { PermissionRow } from './PermissionTabPane/PermissionRow';
+import { List, ListItem } from 'oskari-ui';
+import { withLocale } from 'oskari-ui/util';
 
-export const PermissionsTabPane = (props) => {
+const StyledListItem = styled(ListItem)`
+    &:first-child > div {
+        padding-top: 5px;
+        font-weight: bold;
+    }
+    &:last-child > div {
+        padding-bottom: 20px;
+    }
+`;
+
+const renderRow = (rowModel) => {
     return (
-        <div></div>
+        <StyledListItem>
+            <PermissionRow {...rowModel}/>
+        </StyledListItem>
     );
 };
+
+const PermissionsTabPane = (props) => {
+    const { getMessage, rolesAndPermissionTypes } = props;
+    var permissionDataModel;
+    if (rolesAndPermissionTypes) {
```

Comment:
This is set as required in proptypes so I guess the if isn't needed. Anyway if it IS needed I would refactor to:

  if (!rolesAndPermissionTypes) {
    return;
  }

Reply:
Changed roles and permissions as not mandatory and added if statement as recommended. Earlier these roles and permissions might be undefined if loading from backend is in progress. With recently implemented spinner logic, permission tab should not be rendered with undefined roles and permissions. Still i thought that marking this prop as not required is way to go.